githubapp/client_creator.go

```diff
@@ -219,7 +235,14 @@ func (c *clientCreator) newV4Client(base *http.Client, details string) (*githubv
 		return nil, errors.Wrapf(err, "failed to parse base URL: %q", c.v4BaseURL)
 	}
 
-	client := githubv4.NewEnterpriseClient(v4BaseURL.String(), base)
+	baseClient := base
+	if c.cache != nil {
+		cachedTransport := httpcache.NewTransport(c.cache)
+		cachedTransport.Transport = base.Transport
+		baseClient = cachedTransport.Client()
+	}
+
+	client := githubv4.NewEnterpriseClient(v4BaseURL.String(), baseClient)
```

Comment:
I think using this for the GraphQL API is unnecessary. Only `GET` and `HEAD` requests are cached, and all GraphQL interaction happens via `POST` requests.